crates/wasmtime/src/runtime/component/instance.rs

```diff
@@ -299,9 +305,48 @@ impl Instance {
         store: &StoreOpaque,
         instance: Option<&ComponentExportIndex>,
         name: &str,
-    ) -> Option<ComponentExportIndex> {
+    ) -> Option<(ComponentItem, ComponentExportIndex)> {
         let data = store[self.0].as_ref().unwrap();
         let index = data.component.lookup_export_index(instance, name)?;
+        let ty = match data.component.env_component().export_items[index] {
+            Export::Instance { ty, .. } => TypeDef::ComponentInstance(ty),
+            Export::LiftedFunction { ty, .. } => TypeDef::ComponentFunc(ty),
+            Export::ModuleStatic { ty, .. } | Export::ModuleImport { ty, .. } => {
+                TypeDef::Module(ty)
+            }
+            Export::Type(ty) => ty,
+        };
+        let item = ComponentItem::from(&store.engine(), &ty, &data.ty());
+        Some((
+            item,
+            ComponentExportIndex {
+                id: data.component_id(),
+                index,
+            },
+        ))
+    }
+
+    /// A methods similar to [`Component::get_export_index`] except for this
+    /// instance.
+    ///
+    /// This method will lookup the `name` provided within the `instance`
+    /// provided and return a [`ComponentExportIndex`] which can be passed
+    /// other `get_*` functions like [`Instance::get_func`].
+    ///
+    /// If you need the [`ComponentItem`] corresponding to this export, use
+    /// the [`Instance::get_export`] instead.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `store` does not own this instance.
+    pub fn get_export_index(
```

Comment:
Possible bikeshed: instead of `get_export{,_index}`, maybe `get_export{,_and_type}`? (e.g. favoring the just-the-index version)

Reply:
Eh, maybe, but get_export_and_type feels too smalltalky and 98% of users just use bindgen which hides all that so I'll respectfully decline